src/main/java/mate/academy/spring/controller/ShoppingCartController.java

```diff
@@ -34,14 +36,19 @@ public ShoppingCartController(ShoppingCartService shoppingCartService,
     }
 
     @PutMapping("/movie-sessions")
-    public void addToCart(@RequestParam Long userId, @RequestParam Long movieSessionId) {
+    public void addToCart(Authentication authentication, @RequestParam Long movieSessionId) {
         shoppingCartService.addSession(
-                movieSessionService.get(movieSessionId), userService.get(userId));
+                movieSessionService.get(movieSessionId), getUser(authentication));
     }
 
     @GetMapping("/by-user")
-    public ShoppingCartResponseDto getByUser(@RequestParam Long userId) {
-        User user = userService.get(userId);
+    public ShoppingCartResponseDto getByUser(Authentication authentication) {
+        User user = getUser(authentication);
         return shoppingCartResponseDtoMapper.mapToDto(shoppingCartService.getByUser(user));
     }
+
+    private User getUser(Authentication authentication) {
+        return userService.findByEmail(authentication.getName()).orElseThrow(() ->
+                new EntityNotFoundException("Can't get user by email" + authentication.getName()));
+    }
```

Comment:
Basically, you have copy pasted this code in several places. This is not ok, rather move the repeated code to a common service and use composition (autowire that service where you need it).

Reply:
You can write method `User getUser(String email);` in your UserService